File: 2025_08/leetcode2025-08-01.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<vector<int>> rows;
int init=[](){
    rows = {{1}};
    for (int i=1; i<30; i++){
        vector<int> cur = {1};
        for (int j=0; j<rows[i-1].size()-1; j++){
            cur.push_back(rows[i-1][j]+rows[i-1][j+1]);
        }
        cur.push_back(1);
        rows.push_back(cur);
    }
    return 0;
}();

class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        return vector<vector<int>>(rows.begin(), rows.begin()+numRows);
    }
};

int main(){
    Solution s;
    int numRows = 5;
    vector<vector<int>> res = s.generate(numRows);
    for (auto i : res){
        for (auto j : i){
            cout << j << " ";
        }
        cout << endl;
    }
}

File: leetcode2025/2025_08/leetcode2025-08-01.py
from typing import List

rows = [[1]]
for i in range(1, 30):
    cur = [1]
    for j in range(len(rows[i-1])-1):
        cur.append(rows[i-1][j]+rows[i-1][j+1])
    cur.append(1)
    rows.append(cur)

class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        return rows[:numRows]

if __name__ == "__main__":
    s = Solution()
    numRows = 5
    print(s.generate(numRows))
